### DISCUSSION TITLE
Help with visualising the orientation-dependence of data using an inverse pole figure

### DISCUSSION THREAD

**Discussion by mfuhr-mpg** [mtex-toolbox/mtex#1934] [Ask Anything]
Hello everybody,

I have a probably rather simple task but am not experienced enough with the functions of MTEX to fulfil it.
The task is to plot the orientation-dependent sputtering yield in an inverse pole figure. The orientation data is expressed in the form of Miller indices. Ideally, I want to have a contour plot in order to improve the readability of the plot. What I did so far is the following to get the odf from the Miller indices and attempt to plot the list of sputtering yields Y at the right position in the ipf:

```matlab
h = EN(:,1);
k = EN(:,2);
l = EN(:,3);

ori = Miller(h,k,l,cs);
ori2 = calcDensity(ori,'precision',1*degree);

figure
plot(ori2,Y,'antipodal','contourf')
```

The outcome of the above code is a contour plot of ori2, but Y does not enter the plot. I also tried to use the plotIPDF function but get the error message "Error using plotIPDF: Using plotIPDF with EBSD data is obsolete since MTEX 4.0. Use plotIPDF(ebsd.orientations) instead.", which does not really help since my orientation data does not origin from an EBSD measurement.

Can anyone point out my mistake(s) and help me solve my visualisation task?

Thanks very much in advance!

**Comment by ralfHielscher** [comment]
Hi,

can you explain what the orientation-dependent sputtering yield is? I guess we also need some kind of an formula to relate it to crystal directions.

Ralf.

**Comment by mfuhr-mpg** [comment]
Hi Ralf,

thanks for the quick reply and sorry for the unclear description. The sputtering yields were calculated using SDTrimSP, an algorithm that was developed to describe the interaction of solids with ions. The orientation of the solid was defined using Miller indices. Thus, in short, I have pairs consisting of the orientation of the crystalline (Miller index) and the sputtering yield. Do you have a suggestion for the mentioned visualisation procedure?

Thanks and best regards
Max
________________________________
Von: Ralf Hielscher ***@***.***>
Gesendet: Montag, 28. August 2023 16:53:00
An: mtex-toolbox/mtex
Cc: Fuhr, Maximilian; Author
Betreff: Re: [mtex-toolbox/mtex] Help with visualising the orientation-dependence of data using an inverse pole figure (Discussion #1934)


Hi,

can you explain what the orientation-dependent sputtering yield is? I guess we also need some kind of an formula to relate it to crystal directions.

Ralf.

**Comment by ralfHielscher** [comment]
So your situation is as follows: you have a list of Miller indices stored as
```
hkl = Miller(EN(:,1),EN(:,2),EN(:,3),cs)
```
and you have a corresponding list of sputtering yields
```
sputtingYield = [ your data ]
```
Then you can plot it using
```matlab
hkl = hkl.project2FundamentalRegion;
plot(hkl,data,'fundamentalRegion')
```
Ralf.

**Comment by mfuhr-mpg** [comment]
Thanks for the suggestion – that's exactly the way I also figured out in the meantime.
Do you see any opportunity to turn the plot into a contour plot?
If I use 'contourf' as plotting option, I get the warning `Warning: Contour not rendered for non-finite ZData `.

**Comment by mfuhr-mpg** [comment]
I moved the discussion to another location (https://github.com/mtex-toolbox/mtex/discussions/1435#discussioncomment-6902127) since the error I keep getting is similar to the one of another user. Thanks in advance for any further help!
